unicode-data-names/lib/Unicode/Internal/Char/UnicodeData/NameAliases.hs

```diff
@@ -0,0 +1,439 @@
+-- autogenerated from https://www.unicode.org/Public/14.0.0/ucd/NameAliases.txt
+-- |
+-- Module      : Unicode.Internal.Char.UnicodeData.NameAliases
+-- Copyright   : (c) 2020 Composewell Technologies and Contributors
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+
+{-# LANGUAGE OverloadedLists #-}
+{-# OPTIONS_HADDOCK hide #-}
+
+module Unicode.Internal.Char.UnicodeData.NameAliases
+(NameAliasType(..), NameAliases, nameAliases, nameAliasesList, nameAliasesByType)
+where
+
+import Data.Ix (Ix)
+import Data.List.NonEmpty (NonEmpty, nonEmpty)
+import Data.Semigroup (sconcat)
+import Foreign.C.String (CString)
+import GHC.Exts (Ptr(..))
+
+-- [TODO] @since
+-- | Type of name alias.
+data NameAliasType
+    = Correction
+    -- ^ Corrections for serious problems in the character names.
+    | Control
+    -- ^ ISO 6429 names for C0 and C1 control functions, and other
+    --   commonly occurring names for control codes.
+    | Alternate
+    -- ^ A few widely used alternate names for format characters.
+    | Figment
+    -- ^ Several documented labels for C1 control code points which
+    --   were never actually approved in any standard.
+    | Abbreviation
+    -- ^ Commonly occurring abbreviations (or acronyms) for control codes,
+    --   format characters, spaces, and variation selectors.
+    deriving (Eq, Ord, Enum, Bounded, Ix, Show)
+
+-- [TODO] @since
+-- | Name aliases of a character, if defined.
+{-# INLINE nameAliasesList #-}
+nameAliasesList :: Char -> Maybe (NonEmpty CString)
+nameAliasesList = fmap (sconcat . fmap snd) . nonEmpty . nameAliases
+
+-- [TODO] @since
+-- | Name aliases of a character, if defined.
+{-# INLINE nameAliasesByType #-}
+nameAliasesByType :: NameAliasType -> Char -> Maybe (NonEmpty CString)
+nameAliasesByType t = lookup t . nameAliases
+
+-- [TODO] @since
+-- | A list of aliases of a character
+type NameAliases = [(NameAliasType, NonEmpty CString)]
```

Comment:
A type synonym is generally not very useful, it just adds one more indirection making the code harder to read. We can use the type directly.

Reply:
Done